packages/fuels/src/setup-launch-node-server.ts

```diff
@@ -0,0 +1,142 @@
+    req.on('error', reject);
+  });
+}
+
+const cleanupFns: Map<string, () => Promise<void>> = new Map();
+
+const server = http.createServer(async (req, res) => {
+  res.setHeader('Access-Control-Allow-Origin', '*');
+
+  if (req.url === '/') {
+    try {
+      const body = await parseBody(req);
+
+      const node = await launchNode({
+        port: '0',
+        loggingEnabled: false,
+        debugEnabled: false,
+        ...body,
+        fuelCorePath: 'fuels-core',
+      });
+
+      cleanupFns.set(node.url, async () => {
+        node.cleanup();
+        await waitUntilUnreachable(node.url);
+        cleanupFns.delete(node.url);
+      });
+
+      res.end(node.url);
+    } catch (err) {
+      console.error(err);
+      res.writeHead(500, { 'Content-Type': 'text/plain' });
+      res.end(JSON.stringify(err));
+    }
+    return;
+  }
+
+  if (req.url?.startsWith('/cleanup')) {
+    const nodeUrl = req.url?.match(/\/cleanup\/(.+)/)?.[1];
+    if (nodeUrl) {
+      const cleanupFn = cleanupFns.get(nodeUrl);
+      await cleanupFn?.();
+      res.end();
+    }
+  }
+});
+
+function closeServer() {
+  return new Promise<void>((resolve) => {
+    if (!server.listening) {
+      resolve();
+      return;
+    }
+
+    server.close(async () => {
+      const cleanupCalls: Promise<void>[] = [];
+      cleanupFns.forEach((fn) => cleanupCalls.push(fn()));
+      await Promise.all(cleanupCalls);
+      process.exit();
+    });
+
+    resolve();
+  });
+}
+
+server.on('request', async (req, res) => {
+  if (req.url === '/close-server') {
+    await closeServer();
+    res.end();
+  }
+});
+
+const port = process.argv[2] ? parseInt(process.argv[2], 10) : 49342;
+
+server.listen(port);
+
+server.on('listening', () => {
+  const usedPort = (server.address() as AddressInfo).port;
+  const serverUrl = `http://localhost:${usedPort}`;
+  console.log(serverUrl);
+  console.log(`Server is listening on: ${serverUrl}`);
+  console.log("To launch a new fuel-core node and get its url, make a POST request to '/'.");
+  console.log(
+    "To kill the node, make a POST request to '/cleanup/<url>' where <url> is the url of the node you want to kill."
+  );
+  console.log('All nodes will be killed when the server is closed.');
+  console.log('You can close the server by sending a request to /close-server.');
+});
+
+process.on('exit', () => {
+  console.log('exit');
+  closeServer();
+});
+process.on('SIGINT', () => {
+  console.log('sigint');
+  closeServer();
+});
+process.on('SIGUSR1', () => {
+  console.log('SIGUSR1');
+  closeServer();
+});
+process.on('SIGUSR2', () => {
+  console.log('SIGUSR2');
+  closeServer();
+});
+process.on('uncaughtException', (e) => {
+  console.log('uncaughtException');
+  console.log(e);
+  closeServer();
+});
+process.on('unhandledRejection', (reason) => {
+  console.log('unhandledRejection');
+  console.log(reason);
+
+  closeServer();
+});
+process.on('beforeExit', () => {
+  console.log('beforeExit');
+  closeServer();
+});
```

Comment:
Nit II.

(still, written via the UI)

```suggestion
process.on('exit', closeServer('exit'));
process.on('SIGINT', closeServer('SIGINT'));
process.on('SIGUSR1', closeServer('SIGUSR1'));
process.on('SIGUSR2', closeServer('SIGUSR2'));
process.on('uncaughtException', closeServer('uncaughtException'));
process.on('unhandledRejection', closeServer('unhandledRejection'));
process.on('beforeExit', closeServer('beforeExit'));
```

Reply:
https://github.com/FuelLabs/fuels-ts/pull/2378/commits/9d281fd8a8f3d3e54005b9b6498d51f7749c22a3